eox_core/tests/integration/test_enrollment_integration.py

```diff
@@ -0,0 +1,610 @@
+"""
+Integration tests
+"""
+import json
+from os import environ
+
+import pytest
+import requests
+from django.test import TestCase
+
+
+@pytest.mark.skipif(bool(environ.get('CI')), reason='Do not run on CI')
+class TestEnrollmentIntegration(TestCase):  # pragma: no cover
+    # pylint: disable=too-many-public-methods
+    """Test suite"""
+    data = {}
+
+    @classmethod
+    def setUpClass(cls):
+        with open('eox_core/tests/integration/test_data') as file_obj:
+            cls.data = json.load(file_obj)
+        cls.data['endpoint'] = 'eox-core/api/v1/enrollment/'
+        site1_data = {
+            'client_id': cls.data['site1_data']['client_id'],
+            'client_secret': cls.data['site1_data']['client_secret'],
+            'grant_type': 'client_credentials',
+        }
+        site2_data = {
+            'client_id': cls.data['site2_data']['client_id'],
+            'client_secret': cls.data['site2_data']['client_secret'],
+            'grant_type': 'client_credentials',
+        }
+        request_url = '{}/{}'.format(cls.data['site1_data']['base_url'],
+                                     'oauth2/access_token/')
+        response_site1 = requests.post(request_url, data=site1_data)
+        response_site1.raise_for_status()
+        cls.data['site1_data']['token'] = response_site1.json()['access_token']
+        request_url = '{}/{}'.format(cls.data['site2_data']['base_url'],
+                                     'oauth2/access_token/')
+        response_site2 = requests.post(request_url, data=site2_data)
+        response_site2.raise_for_status()
+        cls.data['site2_data']['token'] = response_site2.json()['access_token']
+
+    @classmethod
+    def tearDownClass(cls):
+        delete_enrollment(cls.data)
+
+    def test_read_valid_email_course(self):
```

Comment:
I think for the test cases maybe we could leave a blank line between the arrange section, the act, and the assert section.
For example :
`
        site1_data = self.data['site1_data']
        data = {
            'username': site1_data['user_id'],
            'course_id': site1_data['course']['id'],
        }
        headers = {
            'Authorization': 'Bearer {}'.format(site1_data['token']),
            'Host': site1_data['host'],
        }
        request_url = '{}/{}'.format(site1_data['base_url'],
                                     self.data['endpoint'])

        response = requests.get(
            request_url,
            data=data,
            headers=headers,
        )

        self.assertEqual(response.status_code, 404)
`

Reply:
So, you want two blocks in each test with the assertion separated from everything else? am i right?

Reply:
I was thinking of three blocks
- The arrange section (where you set up the objects to be tested)
- The act section  (where you call the method/make the request)
- The assert section 
What do you think?

(there is more information about each section in the following [link](https://freecontent.manning.com/making-better-unit-tests-part-1-the-aaa-pattern/))

Reply:
Ok, i will do that.